diff --git a/security/Kconfig b/security/Kconfig
index d9aa521..01fc608 100644
--- a/security/Kconfig
+++ b/security/Kconfig
@@ -230,6 +230,13 @@ config STATIC_USERMODEHELPER_PATH
 	  If you wish for all usermode helper programs to be disabled,
 	  specify an empty string here (i.e. "").

+config SECURITY_LOCKDOWN
+	bool "Enable kernel lockdown"
+	default y
+	help
+	  Enables kernel lockdown which disables any feature that
+	  userspace can abuse to compromise the kernel.
+
 source security/selinux/Kconfig
 source security/smack/Kconfig
 source security/tomoyo/Kconfig
diff --git a/include/linux/security.h b/include/linux/security.h
index 75f4156..a591b90 100644
--- a/include/linux/security.h
+++ b/include/linux/security.h
@@ -402,6 +402,10 @@ void security_inode_invalidate_secctx(struct inode *inode);
 int security_inode_notifysecctx(struct inode *inode, void *ctx, u32 ctxlen);
 int security_inode_setsecctx(struct dentry *dentry, void *ctx, u32 ctxlen);
 int security_inode_getsecctx(struct inode *inode, void **ctx, u32 *ctxlen);
+
+/* Kernel lockdown */
+extern int kernel_lockdown;
+
 #else /* CONFIG_SECURITY */
 struct security_mnt_opts {
 };
diff --git a/security/Makefile b/security/Makefile
index 4d2d378..ccf18e1 100644
--- a/security/Makefile
+++ b/security/Makefile
@@ -30,3 +30,6 @@ obj-$(CONFIG_CGROUP_DEVICE)		+= device_cgroup.o
 # Object integrity file lists
 subdir-$(CONFIG_INTEGRITY)		+= integrity
 obj-$(CONFIG_INTEGRITY)			+= integrity/
+
+# Kernel lockdown
+obj-$(CONFIG_SECURITY_LOCKDOWN)		+= lockdown.o
diff --git a/security/lockdown.c b/security/lockdown.c
new file mode 100644
index 0000000..80ac2fe
--- /dev/null
+++ b/security/lockdown.c
@@ -0,0 +1,4 @@
+// SPDX-License-Identifier: GPL-2.0
+#include <linux/security.h>
+
+int __ro_after_init kernel_lockdown = IS_ENABLED(CONFIG_SECURITY_LOCKDOWN);
